test/server/config_validation/xds_fuzz.cc

```diff
@@ -203,47 +207,59 @@ void XdsFuzzTest::replay() {
       // additional discoveryRequests at launch that we might not want to
       // respond to yet
       EXPECT_TRUE(waitForAck(Config::TypeUrl::get().Listener, std::to_string(version_)));
+      if (removed_name) {
+        modified++;
+        test_server_->waitForCounterGe("listener_manager.listener_modified", modified);
+      } else {
+        added++;
+        test_server_->waitForCounterGe("listener_manager.listener_added", added);
+      }
       break;
     }
     case test::server::config_validation::Action::kRemoveListener: {
-      /* sent_listener = true; */
-      auto removed = removeListener(action.remove_listener().listener_num());
+      auto removed_name = removeListener(action.remove_listener().listener_num());
 
-      if (removed) {
-        updateListener(listeners_, {}, {*removed});
+      if (removed_name) {
+        removed++;
+        updateListener(listeners_, {}, {*removed_name});
         EXPECT_TRUE(waitForAck(Config::TypeUrl::get().Listener, std::to_string(version_)));
+        test_server_->waitForCounterGe("listener_manager.listener_removed", removed);
       }
 
       break;
     }
     case test::server::config_validation::Action::kAddRoute: {
+      if (!sent_listener) {
+        ENVOY_LOG_MISC(info, "Ignoring request to add route_{}", action.add_route().route_num());
+        break;
+      }
       uint32_t route_num = action.add_route().route_num();
-      auto removed = removeRoute(route_num);
+      auto removed_name = removeRoute(route_num);
       auto route = buildRouteConfig(route_num);
       routes_.push_back(route);
 
-      if (removed) {
+      if (removed_name) {
         // if the route was already in routes_, don't send a duplicate add in delta request
         updateRoute(routes_, {}, {});
       } else {
         updateRoute(routes_, {route}, {});
       }
 
-      if (sent_listener) {
-        EXPECT_TRUE(
-            waitForAck(Config::TypeUrl::get().RouteConfiguration, std::to_string(version_)));
-      }
+      EXPECT_TRUE(waitForAck(Config::TypeUrl::get().RouteConfiguration, std::to_string(version_)));
       break;
     }
     case test::server::config_validation::Action::kRemoveRoute: {
-      if (sotw_or_delta_ == Grpc::SotwOrDelta::Sotw) {
-        // routes cannot be removed in SOTW updates
-        break;
-      }
+      // it seems like routes cannot be removed
+      // leaving a route out of an SOTW request does not remove it and sending a
+      // remove message in a delta request is ignored
+      ENVOY_LOG_MISC(info, "Ignoring request to remove route_{}",
+                     action.remove_route().route_num());
```

Comment:
extremely minor nitty question for readability: were the line wraps in this block necessary?

Reply:
the log one goes over the 100 line limit, cleaned up the comment, thanks for catching!